docs/ops/matrix/Einsum_7.md

```diff
@@ -0,0 +1,228 @@
+must be equal in size.
+
+**NOTE**: A label can be repeated in the same input subscript, for example, `equation` equal to `aac,abd,ddde`. In this case, the corresponding dimensions
+must match in size, and the operand is replaced by its diagonal along these dimensions.
+For example, *Einsum* operation on the single 3D tensor of shape `[2, 4, 5, 4]` with `equation` equal to `ijkj->ij`.
+
+**NOTE**: The specification considers the primitive algorithm for *Einsum* operation for better understanding of the operation
+and does not recommend it for implementation.
+
+**NOTE**: The described algorithm can be improved by immediate dimension sum-reduction of the intermediate results if the corresponding labels are absent 
+in the input subscripts of subsequent inputs and the output subscript. It can significantly boost performance and reduce memory costs.
+In the considered example, after the first step you can reduce the dimension corresponding to the label `d`.
+
+The output shape is computed by concatenation of dimension sizes to which labels in the output subscript correspond in the specified order.
+
+Example 1 shows how *Einsum* computes inner product of two 1D tensors:
+
+```
+a1 = [1.0, 2.0, 3.0]
+a2 = [4.0, 5.0, 6.0]
+equation = "i,i->"
+output = 32.0
+```
+
+Example 2 shows how *Einsum* computes matrix-vector multiplication:
+
+```
+A = [[1.0, 2.0, 3.0],
+     [1.0, 2.0, 3.0]]
+b = [4.0, 5.0, 6.0]
+equation = "ij,j->i"
+output = [32.0, 32.0]
+```
+
+Example 3 shows how *Einsum* computes a trace for each batch object:
+
+```
+A = [[[1.0, 2.0, 3.0],
+      [4.0, 5.0, 6.0],
+      [7.0, 8.0, 9.0]],
+     [[2.0, 4.0, 6.0],
+      [8.0, 10.0, 12.0],
+      [14.0, 16.0, 18.0]]]
+equation = "kii->k"
+output = [15.0, 30.0]
+```
+
+Example 4 shows how *Einsum* extracts a diagonal for each batch object:
+
+```
+A = [[[1.0, 2.0, 3.0],
+      [4.0, 5.0, 6.0],
+      [7.0, 8.0, 9.0]],
+     [[2.0, 4.0, 6.0],
+      [8.0, 10.0, 12.0],
+      [14.0, 16.0, 18.0]]]
+equation = "kii->ki"
+output = [[1.0, 5.0, 9.0],
+          [2.0, 10.0, 18.0]]
+```
+
+Example 5 shows how *Einsum* transposes input tensor:
+
+```
+A = [[[1.0, 2.0, 3.0],
+      [4.0, 5.0, 6.0],
+      [7.0, 8.0, 9.0]]]
+equation = "ijk->kij"
+output = [[[1.0, 4.0, 7.0],
+           [2.0, 5.0, 8.0],
+           [3.0, 6.0, 9.0]]]
+```
+
+In addition to a lowercase label, ellipsis `...` can be used as a label in a subscript to cover broadcasted dimensions.
+Each input subscript can contain at most one ellipsis. For example, the ellipsis in input subscript `a...bc` for five rank tensor covers 
+the second and third dimensions. In case input subscripts contain ellipsis for several operands, the dimensions covered by the ellipsis
+must be broadcastable to satisfy numpy broadcasting (or multidirectional broadcasting) rules available in
+<a href="https://numpy.org/doc/stable/user/basics.broadcasting.html#general-broadcasting-rules">numpy broadcasting documentation</a>.
```

Comment:
we have our own document describing broadcasting rules `docs/ops/broadcast_rules.md`

Reply:
Fixed.